docs/auth.md

```diff
@@ -231,3 +307,66 @@ func (t TrustedUserAuth) GetPrincipal(r *http.Request) (Princpal, error) {
     return TrustedUserPrincipal{username: user.Name}, nil
 }
 ```
+
+### Bearer Auth design
+
+Bearer auth is using the [kubernetes apiserver](https://github.com/kubernetes/apiserver) to do delegated authentication and authorization. Kubernetes team will keep this library up to date.
+
+The first thing that you need to do when setting up the generic api server is to tell it the cert, key, port, and address to listen on. You are going to use the [`SecureServingOptions`](https://github.com/kubernetes/apiserver/blob/master/pkg/server/options/serving.go#L38) to set these values.
```

Comment:
I think you should use a permalink for the kubernetes link.

Reply:
I see a handful of other references to master, I think they should all be permalinks.

Source: have read a doc that pointed to master and it was no longer the line that they had intended in the doc.